cmake/SetupCxxFlags.cmake

```diff
@@ -8,6 +8,16 @@ option(OVERRIDE_ARCH "The architecture to use. Default is native." OFF)
 option(ENABLE_SPECTRE_DEBUG "Enable ASSERTs and other SPECTRE_DEBUG options"
   OFF)
 
+# Work around boost not building with clang 15
+# (https://github.com/boostorg/functional/pull/21)
+if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "15.0.0")
+    set(CMAKE_CXX_FLAGS "-D_HAS_AUTO_PTR_ETC=0 ${CMAKE_CXX_FLAGS}")
+  elseif(CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL "15.0.0")
+    set(CMAKE_CXX_FLAGS "-D_HAS_AUTO_PTR_ETC=0 ${CMAKE_CXX_FLAGS}")
```

Comment:
Add this to SetupBoost.cmake please, to the boost target using `target_compile_definitions`.